src/tracks/playabletrack/wavetrack/ui/SpectrumCache.cpp

```diff
@@ -536,7 +517,8 @@ bool WaveClipSpectrumCache::GetSpectrogram(const WaveClip &clip,
    }
 
    // Resize the cache, keep the contents unchanged.
-   mSpecCache->Grow(numPixels, settings, pixelsPerSecond, t0);
+   mSpecCache->Grow(
+      numPixels, settings, samplesPerPixel, t0 /*Grow takes absolute t0*/);
```

Comment:
Almost completing review!  The last thing I must understand is why you changed arguments in the other call of `Grow` but not this one.  Are you correcting an old error?

But I examined things and I find this added comment misleading.  `t0` is, and was before the changes, a time relative to clip start.  Not absolute.  And relative time is in fact what `Grow` was expecting.  The `t0` value is stored in `start` and ultimately used only in `GetSpectrogram` to check for a cache hit.  The `t0` in the comparison is the same `t0` used here.

Reply:
`GetSpectrogram` is called in only one place, and the `t0` value supplied there is from `ClipParameters::t0`, and that value is initialized with a duration relative to clip play start, not an absolute time.